scrapinghub/utils.py

```diff
@@ -105,3 +107,28 @@ def format_iter_filters(params):
         if filter_data:
             params['filter'] = filter_data
     return params
+
+
+def parse_auth(auth):
+    """Parse authentification token.
+
+    >>> parse_auth(None)
+    >>> parse_auth(('user', 'pass'))
+    ('user', 'pass')
+    >>> parse_auth('user:pass')
+    ('user', 'pass')
+    >>> parse_auth('apikey')
+    ('apikey', '')
+    >>> parse_auth('312f322f333a736f6d652e6a77742e746f6b656e')
+    ('1/2/3', 'some.jwt.token')
+    """
+    if auth is None or isinstance(auth, tuple):
+        return auth
+    try:
+        auth = decode(auth, 'hex_codec')
+        if not isinstance(auth, string_types):
+            auth = auth.decode('ascii')
+    except binascii.Error:
+        pass
```

Comment:
@vshlapakov did you try it with real apikey? it won't work correctly because real apikey consists of valid hexademical characters. you should try to decode the apikey, split it in parts, check that username part is a valid jobkey and password part is not empty - and only after that you can assume that this is a jwt token auth and can be used for auth. 

another option is to check apikey len, but this wouldn't allow us to change apikey len in future wihtout causing users to upgrade